### PR TITLE
Add import for CA before sending request to registry

### LINKED ISSUE #583: Custom CA certificates not fully accepted
I'm getting errors when deploying Dynatrace operator in Dynatrace Managed environments where the Dynatrace environment has a certificate issued by a custom certificate authority.

Steps to replicate:

1. Deploy Dynatrace Operator
```
kubectl create namespace dynatrace
kubectl apply -f https://github.com/Dynatrace/dynatrace-operator/releases/download/v0.4.2/kubernetes.yaml
```

2. Deploy DynaKube
```
kubectl -n dynatrace create secret generic dynakube --from-literal="apiToken=<apitoken>"
kubectl create configmap --from-file certs ambitca -n dynatrace
kubectl apply -f classicFullStack.yaml
```
where the file named certs contains valid CA certificate of the CA which issued the certificate for the Dynatrace Managed environment.
classicFullStack.yaml contains:
```
apiVersion: dynatrace.com/v1beta1
kind: DynaKube
metadata:
  name: dynakube
  namespace: dynatrace
spec:
  apiUrl: https://dynatrace-managed-hostname/e/09e3c1e7-3b21-4488-a852-2d389965abd7/api
  trustedCAs: ambitca
  oneAgent:
    classicFullStack:
      image: ""
      tolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/master
          operator: Exists
```

I can confirm the managed environment certificate is trusted by running
```
curl --cacert certs https://dynatrace-managed-hostname
```
3. The operator log contains the following exceptions:
```
{"error":"failed to get image version: pinging container registry dynatrace-managed-hostname: Get \"https://dynatrace-managed-hostname/v2/\": x509: certificate signed by unknown authority","level":"error","logger":"dynakube-updates","msg":"failed to update OneAgent image version","stacktrace":"Get \"https://dynatrace-managed-hostname/v2/\": x509: certificate signed by unknown authority
pinging container registry dynatrace-managed-hostname
github.com/containers/image/v5/docker.(*dockerClient).detectPropertiesHelper
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_client.go:759
github.com/containers/image/v5/docker.(*dockerClient).detectProperties.func1
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_client.go:792
sync.(*Once).doSlow
        /usr/local/go/src/sync/once.go:68
sync.(*Once).Do
        /usr/local/go/src/sync/once.go:59
github.com/containers/image/v5/docker.(*dockerClient).detectProperties
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_client.go:792
github.com/containers/image/v5/docker.(*dockerClient).makeRequest
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_client.go:462
github.com/containers/image/v5/docker.(*dockerImageSource).fetchManifest
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_image_src.go:195
github.com/containers/image/v5/docker.(*dockerImageSource).ensureManifestIsLoaded
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_image_src.go:229
github.com/containers/image/v5/docker.newImageSourceAttempt
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_image_src.go:133
github.com/containers/image/v5/docker.newImageSource
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_image_src.go:75
github.com/containers/image/v5/docker.dockerReference.NewImageSource
        /go/pkg/mod/github.com/containers/image/v5@v5.19.0/docker/docker_transport.go:144
github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/dtversion.GetImageVersion
        /app/src/controllers/dynakube/dtversion/docker.go:40
github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/updates.updateImageVersion
        /app/src/controllers/dynakube/updates/version.go:81
github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/updates.ReconcileVersions
        /app/src/controllers/dynakube/updates/version.go:63
github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube.(*ReconcileDynaKube).reconcileDynaKube
        /app/src/controllers/dynakube/dynakube_controller.go:200
github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube.(*ReconcileDynaKube).Reconcile
        /app/src/controllers/dynakube/dynakube_controller.go:125
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1581
```

The k8s cluster itself trusts the custom certificate and the operator is able to pull images, however the operator is unable to fetch new OneAgent versions and does not update the OneAgent pods.

The full [Dynatrace Operator log](https://github.com/Dynatrace/dynatrace-operator/files/8083174/operator.log)

### PR DESCRIPTION
**Pull request by gkrenn**
Fixes the bug that custom certificates were not taken into account when sending requests to the image registry.

Resolves #583

### CODE REVIEW

File: config/helm/chart/default/templates/Common/operator/deployment-operator.yaml [hunk 1]
```diff
@@ -94,6 +97,9 @@ spec:
                     operator: In
                     values:
                       - linux
+      volumes:
+        - emptyDir: { }
+          name: tmp-dir
```
Comment (0sewa0) [inline config/helm/chart/default/templates/Common/operator/deployment-operator.yaml:102]: nitpick: call it something more descriptive, we will forget what this tmp-dir does in 2 weeks :D
maybe `tmp-cert-dir`

File: src/controllers/dynakube/dtclient_builder.go [hunk 2]
```diff
@@ -34,8 +35,7 @@ type DynatraceClientProxy struct {
 }
 
 const (
-	proxy        = "proxy"
-	certificates = "certs"
+	proxy = "proxy"
```
Comment (0sewa0) [inline src/controllers/dynakube/dtclient_builder.go:38]: wouldn't these const fit better in the `dtclient` package ?

File: src/controllers/dynakube/dtversion/docker.go [hunk 3]
```diff
@@ -13,7 +13,12 @@ import (
 )
 
 // VersionLabel is the name of the label used on ActiveGate-provided images.
-const VersionLabel = "com.dynatrace.build-version"
+const (
+	VersionLabel                   = "com.dynatrace.build-version"
+	TmpCAPath                      = "/tmp/dynatrace-operator"
+	TmpCAName                      = "dynatraceCustomCA.crt"
+	CustomCertificatesConfigMapKey = "certs"
```
Comment (0sewa0) [inline src/controllers/dynakube/dtversion/docker.go:20]: wouldn't `CustomCertificatesConfigMapKey ` fit better in the `dtclient` package ?

File: src/controllers/dynakube/dtclient_builder.go (hunk 2 above)
Comment (gkrenn) [reply src/controllers/dynakube/dtclient_builder.go]: moved

File: src/controllers/dynakube/dtversion/docker.go (hunk 3 above)
Comment (gkrenn) [reply src/controllers/dynakube/dtversion/docker.go]: moved

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**